### ISSUE TITLE
[Discussion] Jinja usage for zone data dict vs ansible `vars`

### ISSUE THREAD

**Issue by nE0sIghT** [ganto/ansible-gdnsd#28]
Hello!

First of all thanks for your project - your ideas (from blog post [1]) related to zonefile handling are used in several projects.

Our team had some discussion about your way of data handling in Jinja templates:

```
{% set _zone_data = {} %}
{% set _ = _zone_data.update({'ttl': gdnsd__include_zone.ttl|d(gdnsd__ttl)}) %}
```

It was written 8 years ago and syntactically not obvious on 1st eye tbh.

Just wonder what do you think about using `vars` to handle dict in the Ansible task?

Eg

```
- name: '{{ gdnsd__include_zone.domain }} - Generate forward zone file'
  ansible.builtin.template:
    src: 'zone.j2'
    ...
  vars:
    _zone_data:
      ttl: "gdnsd__include_zone.ttl|d(gdnsd__ttl)"
      ...
```

[1] https://linuxmonk.ch/wordpress/index.php/2016/managing-dns-zones-with-ansible/

**Comment by ganto** [comment]
Thanks a lot for reading my code and giving me feedback. 

To be honest I've never really though of using `vars` in an Ansible task for declaring template-internal "jinja" variables. What would you gain from it? Would you say it's more readable? 

I'm not exactly sure what I should think of this suggestion. Something I don't really like on first sight is that it would split up the responsibility to define all the necessary subkeys to two different places (the `vars` and the template itself).

On the other hand I guess that's just a question of style.

**Comment by nE0sIghT** [comment]
Thanks for quick reply!

Yes, this is fully a style question. In our team we thinks about Ansible as something like MVC system. So for us inventory is `Model`, templates are `Views` and tasks are `Controler`.
So "it would split up the responsibility" is exactly what we gain with it as `S` in `SOLID` - inventory will define data, tasks will manipulate them and templates will render them.

In that particular case we had discussion if `set _ = _zone_data.update` construct is ever acceptable for our code style from the perspective of readability and single responsibility principles.

Actually we discussed another project [1] which inherits principles from your blog post. Templates in that project looks heavy so we finally decided to move data manipulation from templates to task.

Anyway thanks for sharing your view on that - it's valuable to hear from people whose code and ideas impacts the world.

[1] https://github.com/bertvv/ansible-role-bind/blob/master/templates/reverse_zone.j2
